python/datafusion/context.py

```diff
@@ -0,0 +1,1167 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+from ._internal import SessionConfig as SessionConfigInternal
+from ._internal import RuntimeConfig as RuntimeConfigInternal
+from ._internal import SQLOptions as SQLOptionsInternal
+from ._internal import SessionContext as SessionContextInternal
+from ._internal import LogicalPlan, ExecutionPlan  # TODO MAKE THIS A DEFINED CLASS
+
+from datafusion._internal import AggregateUDF
+from datafusion.catalog import Catalog, Table
+from datafusion.dataframe import DataFrame
+from datafusion.expr import Expr
+from datafusion.record_batch import RecordBatchStream
+from datafusion.udf import ScalarUDF
+
+from typing import Any, TYPE_CHECKING
+from typing_extensions import deprecated
+
+if TYPE_CHECKING:
+    import pyarrow
+    import pandas
+    import polars
+
+
+class SessionConfig:
+    def __init__(self, config_options: dict[str, str] = {}) -> None:
```

Comment:
It's generally best to avoid mutable defaults: https://docs.astral.sh/ruff/rules/mutable-argument-default/

Reply:
Thanks! Updaed.